### PR TITLE
Toggle series visibility from chart legend (pie charts)

### LINKED ISSUE #2697: Toggle series on/off in multiple series charts
Can we make it so that clicking on the legend for a series toggles it on or off in the chart (and the chart axis adjusts accordingly).

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

### PR DESCRIPTION
**Pull request by kulyk**
Implements #2697 for pie charts ([product doc](https://www.notion.so/metabase/Let-users-hide-show-series-by-clicking-on-the-legend-90a5ec138b534f039136ea7f377252b8?pvs=4)) and effectively closes #2697

Allows to toggle series visibility from the chart legend by clicking the legend item dot (FWIW, series visibility set like that isn't meant to be persisted). Also, updates the `pie.rows` visualization setting widget to better distinguish its "hide" action from the new legend behavior (similarly to `graph.series_order` in cartesian charts)

### To verify

1. Create a multi-series chart (e.g., Count of Orders per Product Category)
2. Ensure you can click the legend dots to show/hide the slice
3. Ensure you can't hide the last slice
4. Ensure you can hide the "Other" slice
5. In chill mode, ensure you can manage slice order and related settings via the "Data" tab

### Demo

https://github.com/user-attachments/assets/b0a32d8c-0d3c-442a-98a4-40e973469dcb

### CODE REVIEW

File: frontend/src/metabase/visualizations/echarts/pie/tooltip.tsx [hunk 1]
```diff
@@ -33,16 +33,25 @@ export const getTooltipOption = (
     ...getTooltipBaseOption(containerRef),
     trigger: "item",
     formatter: params => {
-      if (Array.isArray(params) || typeof params.dataIndex !== "number") {
+      if (Array.isArray(params) || !params.name) {
+        return "";
+      }
+
+      const slice =
```
Comment (alxnddr) [inline frontend/src/metabase/visualizations/echarts/pie/tooltip.tsx:40]: It is a corner case, but would be nice to fix by switching from using slice names as identifiers:
https://github.com/user-attachments/assets/de74a074-c6ec-4909-89fd-19a1c01679d2

**Review by alxnddr — COMMENTED** [review]
Looks great! I left one comment with a corner case that would be nice to fix, please have a look

File: frontend/src/metabase/visualizations/echarts/pie/tooltip.tsx (hunk 1 above)
Comment (kulyk) [reply frontend/src/metabase/visualizations/echarts/pie/tooltip.tsx]: Good catch, thanks! [`56036a8`](https://github.com/metabase/metabase/pull/47581/commits/56036a8b398edb5d69c69ec8f1d5b3536f30e29a) should fix it, I found a way to make it work using the `dataIndex`

Also, I noticed that if there's a real pie slice called "Other" + the grouped "Other" slice, the toggle behavior wasn't working as expected. The problem was that we use the `slice.data.key` property to keep the list of hidden slices, which happened to be the same in these two scenarios. In [`17a4c1f`](https://github.com/metabase/metabase/pull/47581/commits/17a4c1fdc2b481d807f025b25a980a8193ea6e98) I've reworked it so the grouped "Other" slice's key is `___OTHER___` (cc @EmmadUsmani)

**Review by alxnddr — APPROVED** [review]

**Review by kdoh — CHANGES_REQUESTED** [review]
Looking nice overall. Found a couple wonky things using it.

Screenshot 2024-09-06 at 7 36 24 AM

The bubbles and labels on the branch as of this morning (`eb0831cdaf523e280a5a5f9d4af3134522a5b18d`) were misaligned as you can see here. They should be center aligned horizontally like they are in the Cartesian legend.

I'm also noticing a bit of funk when I hide certain series. In [this example](https://www.loom.com/share/09d1737d1b824f58a3eeef3821d7719c) when I hide `Doohickey` the legend jumps around which makes the thing feel sort of unstable. I think it's important that the legend to never move around as you're interacting with it.

Other than those couple of things it feels really nice!

**Comment by kulyk** [comment]
@kdoh, thanks for the review, everything should be addressed now, please take another look 🙏 

https://github.com/user-attachments/assets/442d5cb6-c9b4-49bb-858b-8464f4e127ee

**Comment by kdoh** [comment]
@kulyk The jumping is fixed but I'm still seeing alignment issues. Just checked on both Firefox Dev (my daily browser) and Safari and seeing the spacing issue still on my machine.

Screenshot 2024-09-06 at 5 08 26 PM

**Comment by kulyk** [comment]
@kdoh, the alignment should be fixed now, please take a look 🙏 Thanks!

CleanShot 2024-09-09 at 16 20 47@2x

**Review by kdoh — APPROVED** [review]
Looking good now.

**Pull request merged**